src/main/java/org/cryptomator/ui/vaultoptions/GeneralVaultOptionsController.java

```diff
@@ -23,36 +26,62 @@
 
 	private final Stage window;
 	private final Vault vault;
+	private final VaultSettings vaultSettings;
 	private final HealthCheckComponent.Builder healthCheckWindow;
 	private final ResourceBundle resourceBundle;
 
 	public TextField vaultName;
 	public CheckBox unlockOnStartupCheckbox;
+	public CheckBox filenameLengthRestrictionCheckbox;
+	public NumericTextField filenameLengthField;
 	public ChoiceBox<WhenUnlocked> actionAfterUnlockChoiceBox;
 
 	@Inject
 	GeneralVaultOptionsController(@VaultOptionsWindow Stage window, @VaultOptionsWindow Vault vault, HealthCheckComponent.Builder healthCheckWindow, ResourceBundle resourceBundle) {
 		this.window = window;
 		this.vault = vault;
+		this.vaultSettings = vault.getVaultSettings();
 		this.healthCheckWindow = healthCheckWindow;
 		this.resourceBundle = resourceBundle;
 	}
 
 	@FXML
 	public void initialize() {
-		vaultName.textProperty().set(vault.getVaultSettings().displayName().get());
-		vaultName.focusedProperty().addListener(this::trimVaultNameOnFocusLoss);
+		vaultName.textProperty().set(vaultSettings.displayName().get());
+		vaultName.focusedProperty().addListener(this::trimAndSetVaultNameOnFocusLoss);
 		vaultName.setTextFormatter(new TextFormatter<>(this::removeWhitespaces));
-		unlockOnStartupCheckbox.selectedProperty().bindBidirectional(vault.getVaultSettings().unlockAfterStartup());
+
+		unlockOnStartupCheckbox.selectedProperty().bindBidirectional(vaultSettings.unlockAfterStartup());
+
 		actionAfterUnlockChoiceBox.getItems().addAll(WhenUnlocked.values());
-		actionAfterUnlockChoiceBox.valueProperty().bindBidirectional(vault.getVaultSettings().actionAfterUnlock());
+		actionAfterUnlockChoiceBox.valueProperty().bindBidirectional(vaultSettings.actionAfterUnlock());
 		actionAfterUnlockChoiceBox.setConverter(new WhenUnlockedConverter(resourceBundle));
+
+		int maxClearTextFileNameLength = vaultSettings.maxCleartextFilenameLength().get();
+		filenameLengthRestrictionCheckbox.selectedProperty().set(maxClearTextFileNameLength != Constants.UNKNOWN_CLEARTEXT_FILENAME_LENGTH_LIMIT && maxClearTextFileNameLength != Constants.UNLIMITED_CLEARTEXT_FILENAME_LENGTH);
+		filenameLengthRestrictionCheckbox.selectedProperty().addListener(this::resetFileNameLimit);
+
+		filenameLengthField.setText(maxClearTextFileNameLength);
+		filenameLengthField.focusedProperty().addListener(this::setFileNameLengthLimitOnFocusLoss);
```

Comment:
* either use "listener"-style names such as `filenameLengthFieldFocusDidChange`, not "action" names
* or reference action from fxml (`onFocus="#applyLimit"`)